### PR TITLE
Manacher's Algorithm - C++

### PR DESCRIPTION
**Pull request by SangeetaMishra**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#667

### CODE REVIEW

**Comment by SangeetaMishra** [comment]
@iamrajiv @Abhijit2505 @atarax665 @Kajol-Kumari  Can you please review this PR.

File: C-Plus-Plus/search/Manacher's Algorithm.cp [hunk 1]
```diff
@@ -0,0 +1,85 @@
+#include <bits/stdc++.h>
```
Comment (iamrajiv) [inline C-Plus-Plus/search/Manacher's Algorithm.cp]: @SangeetaMishra Add a description of Manacher's Algorithm.

File: C-Plus-Plus/search/Manacher's Algorithm.cp [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            ans=max(ans, 2*x);
+            cout<<i-x<<" "<<i+x-1<<"\n";
+            x--;
+        }
+    }
+    cout<<"Length of longest palindromic substring: "<<ans;
+    return 0;
+}
+
+/*
```
Comment (iamrajiv) [inline C-Plus-Plus/search/Manacher's Algorithm.cp]: @SangeetaMishra Add time and space complexity.

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@SangeetaMishra Remove whitespace from file name because every character is treated as hexadecimal value and sometimes it gets converted to hexadecimal values for some character. For whitespace, it's hexadecimal value is `20` that's why in `C-Plus-Plus/README.md` the file name is converted to `Manacher's%20Algorithm.cp`.

**Review by iamrajiv — APPROVED** [review]

**Review by atarax665 — APPROVED** [review]

**Comment by SangeetaMishra** [comment]
@Abhijit2505  Please review the PR

**Review by vatsalkesarwani12 — APPROVED** [review]
able to merge

**Pull request merged**